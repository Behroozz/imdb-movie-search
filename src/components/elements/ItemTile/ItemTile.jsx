import React from 'react'
import { get } from 'lodash'

const ItemTile = ({ item }) => {
  const smallImage = item.image || 'https://via.placeholder.com/400'
  const medImage = item.image || 'https://via.placeholder.com/640'
  const largeImage = item.image || smallImage
  return (
    <div className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2">
      <div className="bg-white h-full py-2 px-4">
        <h3 className="text-xl">{get(item, 'Title')}</h3>
        <h4>Year : {get(item, 'Year')}</h4>
        <img className="w-full" sizes="" srcSet={`${smallImage}, ${medImage}, ${largeImage}`} />
        <a className="mt-4 text-blue-500 block" target="_blank" href={`https://imdb.com/title/${get(item, 'imdbID')}`}>See more information on {get(item, 'Title')}</a>
      </div>
    </div>
  )
}

export default React.memo(ItemTile)