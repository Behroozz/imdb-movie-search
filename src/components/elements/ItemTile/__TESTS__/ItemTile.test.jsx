import React from 'react'
import renderer from 'react-test-renderer'
import ItemTile from '../ItemTile.jsx'

const tileMock = {
  Title: "Waterworld",
  Year: 1995,
  imdbID: "tt0114898"
}

test('should render a single item tile', () => {
  const component = renderer.create(
    <ItemTile item={tileMock} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})