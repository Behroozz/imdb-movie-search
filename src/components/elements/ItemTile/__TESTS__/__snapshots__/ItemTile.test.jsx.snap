// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single item tile 1`] = `
<div
  className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
>
  <div
    className="bg-white h-full py-2 px-4"
  >
    <h3
      className="text-xl"
    >
      Waterworld
    </h3>
    <h4>
      Year : 
      1995
    </h4>
    <img
      className="w-full"
      sizes=""
      srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
    />
    <a
      className="mt-4 text-blue-500 block"
      href="https://imdb.com/title/tt0114898"
      target="_blank"
    >
      See more information on 
      Waterworld
    </a>
  </div>
</div>
`;
